src/main/java/com/box/sdk/BoxSearch.java

```diff
@@ -0,0 +1,55 @@
+package com.box.sdk;
+import java.net.URL;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+/**
+ * Represents search on Box. This class can be used to search through your box instance.
+ * In addition this lets you take advantage of all the advanced search features.
+ *
+ * <p>Unless otherwise noted, the methods in this class can throw an unchecked {@link BoxAPIException} (unchecked
+ * meaning that the compiler won't force you to handle it) if an error occurs. If you wish to implement custom error
+ * handling for errors related to the Box REST API, you should capture this exception explicitly.</p>
+ */
+public class BoxSearch extends BoxResource {
```

Comment:
I don't think this class needs to extend BoxResource. BoxResource objects are typically ID'd items that can be returned by the API.